docs/specification/2025-03-26/basic/authorization.md

```diff
@@ -4,242 +4,391 @@ type: docs
+document path, following 
+[RFC8414](https://datatracker.ietf.org/doc/html/rfc8414#section-3).
 
-**NOTE**: The following example assumes the MCP server is also functioning as the
-authorization server. However, the authorization server may be deployed as its own
-distinct service.
+An MCP client may use the `registration_endpoint` to register itself with OAuth Provider. An MCP 
+client may also be pre-registered with certain OAuth Providers.
 
-A human user completes the OAuth flow through a web browser, obtaining an access token
-that identifies them personally and allows the client to act on their behalf.
+An MCP client **MUST** check the `resource` identifier against the hostname of the MCP server. An MCP 
+client **MUST** send the `resource` identifier to the OAuth Provider, following 
+[RFC8707](https://www.rfc-editor.org/rfc/rfc8707.html).
 
-When authorization is required and not yet proven by the client, servers **MUST** respond
-with _HTTP 401 Unauthorized_.
+An MCP client **SHOULD** implement Demonstrating Proof of Possession (DPoP) and use it to the extent 
+supported by the OAuth Provider. See 
+[RFC9449](https://datatracker.ietf.org/doc/html/rfc9449).
 
-Clients initiate the
-[OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12#name-authorization-code-grant)
-authorization flow after receiving the _HTTP 401 Unauthorized_.
+Clients initiate the 
+[OAuth 2.1 IETF DRAFT](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-12) authorization 
+flow after receiving the *HTTP 401 Unauthorized*.
 
-The following demonstrates the basic OAuth 2.1 for public clients using PKCE.
+The following demonstrates the OAuth 2.1 flow for public clients using PKCE.
 
 ```mermaid
 sequenceDiagram
-    participant B as User-Agent (Browser)
-    participant C as Client
-    participant M as MCP Server
-
-    C->>M: MCP Request
-    M->>C: HTTP 401 Unauthorized
-    Note over C: Generate code_verifier and code_challenge
-    C->>B: Open browser with authorization URL + code_challenge
-    B->>M: GET /authorize
-    Note over M: User logs in and authorizes
-    M->>B: Redirect to callback URL with auth code
-    B->>C: Callback with authorization code
-    C->>M: Token Request with code + code_verifier
-    M->>C: Access Token (+ Refresh Token)
-    C->>M: MCP Request with Access Token
-    Note over C,M: Begin standard MCP message exchange
+    participant UA as User-Agent (Browser or App/Broker)
+    participant MC as MCP Client
+    participant MS as MCP Server
+    participant IDP as Identity Provider
+
+    MC->>MS: MCP Request
+    MS-->>MC: 401 Unauthorized with issuer, scope, and resource in WWW-Authenticate
+    MC->>MC: Construct OIDC /authorize endpoint from issuer identifier
+    Note over MC: Generate code_verifier and code_challenge,<br/>and request with scope and resource
+    MC-->>UA: Open /authorize
+    Note over UA: User authenticates to the IDP
```

Comment:
```suggestion
    UA-->>IDP: User authenticates
```

Reply:
^ Explicitly describes that this is an interaction between the User-Agent and the IDP